features/update/async_accepted/feature.cs

```diff
@@ -0,0 +1,92 @@
+namespace update.async_accepted;
+
+using Temporalio.Activities;
+using Temporalio.Client;
+using Temporalio.Exceptions;
+using Temporalio.Features.Harness;
+using Temporalio.Worker;
+using Temporalio.Workflows;
+using Xunit;
+
+class Feature : IFeature
+{
+    [Workflow]
+    class MyWorkflow
+    {
+        private bool shutdown;
+        private bool finishUpdate;
+
+        [WorkflowRun]
+        public Task RunAsync() => Workflow.WaitConditionAsync(() => shutdown);
+
+        [WorkflowSignal]
+        public async Task ShutdownAsync() => shutdown = true;
+
+        [WorkflowSignal]
+        public async Task FinishUpdateAsync() => finishUpdate = true;
+
+        [WorkflowUpdate]
+        public async Task<int> SuccessfulUpdateAsync()
+        {
+            await Workflow.WaitConditionAsync(() => finishUpdate);
+            finishUpdate = false;
```

Comment:
I'm a bit skeptical about setting this to `false` on exit, since it would affect other in-flight updates. (Obviously it works for the code you have, I'm just bearing in mind that we're partly aiming for this repo to contain good example code for public readers.)

Reply:
This workflow does not support multiple in-flight updates. It is normal for people to code their workflows to expected assumptions/constraints. This workflow is specifically for the code that uses it. Many of our feature workflows are developed for certain expectations but not hypothetical other ones. I don't think it's reasonable to review the feature workflows as a general purpose ones.